examples/data-objects/pond/src/index.tsx

```diff
@@ -30,102 +37,125 @@ export const PondName = "Pond";
-        div.style.border = "1px dotted green";
-        div.style.padding = "5px";
-
-        const title = document.createElement("h1");
-        title.innerText = "Pond";
-
-        const index = document.createElement("h4");
-        index.innerText =
-            `dotted borders denote different component boundaries`;
-
-        div.appendChild(title);
-        div.appendChild(index);
-
-        // Sub-Component setup
-        const clicker2Div = document.createElement("div");
-        const clicker3Div = document.createElement("div");
-        div.appendChild(clicker2Div);
-        div.appendChild(clicker3Div);
-
-        this.clickerView.render(clicker2Div);
-        this.clickerUsingProvidersView.render(clicker3Div);
-
-        return div;
+        this._exampleUsingProviders = await exampleUsingProvidersHandle.get();
     }
 
-    // end IFluidHTMLView
-
     // ----- COMPONENT SETUP STUFF -----
 
     public static getFactory() { return Pond.factory; }
 
-    private static readonly factory = new DataObjectFactory(
+    public static readonly factory = new DataObjectFactory(
         PondName,
         Pond,
         [SharedDirectory.getFactory()],
         {},
         new Map([
-            Clicker.getFactory().registryEntry,
+            DoubleCounter.getFactory().registryEntry,
             ExampleUsingProviders.getFactory().registryEntry,
         ]),
     );
 }
 
+interface IPondViewProps {
+    model: Pond;
+}
+
+const PondView: React.FC<IPondViewProps> = (props: IPondViewProps) => {
+    const { model } = props;
+    return (
+        <>
+            <h1>Pond</h1>
+            <h4>dotted borders denote different component boundaries</h4>
+            <DoubleCounterView counter1={ model.doubleCounter.counter1 } counter2={ model.doubleCounter.counter2 } />
+            <ExampleUsingProvidersView userInfo={ model.exampleUsingProviders.userInformation } />
+        </>
+    );
+};
+
 // ----- CONTAINER SETUP STUFF -----
 
 const dependencyContainer = new DependencyContainer<FluidObject<IFluidUserInformation>>();
 dependencyContainer.register(IFluidUserInformation, async (dc) => userInfoFactory(dc));
 
-export const fluidExport = new ContainerRuntimeFactoryWithDefaultDataStore(
-    Pond.getFactory(),
-    new Map([
-        Pond.getFactory().registryEntry,
-    ]),
-    dependencyContainer,
-);
+const dataStoreId = "modelDataStore";
+
+// This request handler responds to the default request by pairing the default Pond model with a PondView.
+const pondViewRequestHandler = async (request: RequestParser, runtime: IContainerRuntime) => {
+    if (request.pathParts.length === 0) {
+        const objectRequest = RequestParser.create({
+            url: ``,
+            headers: request.headers,
+        });
+        const fluidObject = await requestFluidObject<Pond>(
+            await runtime.getRootDataStore(dataStoreId),
+            objectRequest);
+        const viewResponse = <PondView model={ fluidObject } />;
```

Comment:
From the consumer perspective this view finally ends up being generic view that you cannot compose, pass props etc. If I created two separate Pond Views though this request - the only way you could exchange data between them would be through common data object or consumer/provider pattern?

Reply:
With this setup, if you requested two `PondView`s they would share the same common `Pond` so the data would be shared.  If you wanted multiple `Pond`s you would need to create a separate container or rewrite the `containerInitializingFirstTime()`, and then yes you'd need to set up some different means of communication depending on what you wanted to share.

Here `PondView` plays a role of an app/container view more than a data object view really, and `Pond` plays a role of "overall app data" more than a specific, granular, generally useful collection of data.  So their design doesn't really worry too much about generic usage or sharability of these since they're custom-tweaked for the demo.